### DISCUSSION TITLE
decorative rings around hit-circles and sliders

### DISCUSSION THREAD

**Discussion by iSimpFor** [ppy/osu#16885] [Ideas]
the idea of this is to have decorative rings around circles and sliders which mappers can use

instead of new combos and combo coloring, mappers can use them to emphasize notes; indicate a rhythm change; or just make their map look nicer

there can be multiple different decorative rings to use differently, depending on how the mappers wants to use them

the approach circle will appear on top of the ring and the ring will not look like the approach circle to prevent confusion 

for sliders
-the decoration will cover the whole slider body
-if multiple parts of a slider are marked with a decoration, the decorations can merge together
-decorations on the same reverse circle can have a effect where they can appear or disappear when next hit

these decorative rings will appear regardless of storyboard/video settings

examples:
![ring](https://user-images.githubusercontent.com/44514667/154201423-f897e1d0-917f-43f1-a8a5-8111265a4ba5.png)
![design](https://user-images.githubusercontent.com/44514667/154202283-83551dc5-0d37-49a0-807a-1ad9ae21ff9d.png)
![sliders](https://user-images.githubusercontent.com/44514667/154203193-65a92de9-9808-4bba-b9f3-c12cdf3ea28a.png)
![mix](https://user-images.githubusercontent.com/44514667/154204049-5bf749b2-1eb4-4fe5-baa2-2ac1dd8fdfbe.png)

**Comment by peppy** [comment]
Sounds like per object skinning if anything. If you're proposing that this is used to show bpm changes, then it would be implemented at a system level, otherwise it would just be a new skin element. Seems you're probably proposing as stylistic only.

**Reply by iSimpFor** [reply]
in a way, describing it as per object skinning is true as you're giving an object a different decoration but i don't think it would literally be a different skin per decorative circle(depending on how this is implemented) 

i think these would be applied like how you apply hitsounds to an object(with the sound toggles and you can import custom rings and there is a set of them)

i believe that they can be used stylisticly and with a purpose, depending on what the mapper wants to do with them(and they can use it more than just for bpm changes)

i do agree that this would probably be a new set of skin elements however
